PowerEditor/src/tools/NppModernShell/README.md

```diff
@@ -0,0 +1,61 @@
+# Windows 11 Modern UI Context Menu integration
+
+The purpose of this project is to allow Notepad++ to integrate into the new Windows 11 right-click context menu.
+Doing this requires two new things.
+
+* A dll library with some COM objects that the shell can communicate with.
+* A Sparse Package containing the metadata for the COM server.
+
+To build this, the following steps needs to be taken:
+
+1. Build a Release dll file (NppModernShell.dll)
+2. Generate a Sparse Package (NppModernShell.msix)
+3. Sign both of these with signtool.exe
+4. Make sure they are included in the installer, so they are deployed next to the notepad++.exe program.
+5. The installer should, upon installation, install the package.
+6. The installer should, upon uninstallation, uninstall the package.
+
+## Build a Release dll file (NppModernShell.dll)
+Just open the NppModernShell.sln Visual Studio solution, select Release as the build type, and do a Rebuild of the solution.
+
+## Generate a Sparse Package (NppModernShell.msix)
+To generate a Sparse Package, you need to have the makeappx.exe tool in your PATH, the easiest way to do this is to run the `Developer Command Prompt for VS 2022` command prompt, since it sets up the path.
+Once inside the NppModernShell folder, run the following command to generate the Sparse Package:
+```
+makeappx pack /d .\Packaging /p .\NppModernShell.msix /nv
+```
+This takes the content of the Packaging directory, and packages them up into the msix file.
+
+## Sign both of these with signtool.exe
+Now we have both the `NppModernShell.dll` and `NppModernShell.msix` files, we need to sign them with a valid certificate.
+To do this, once again run the `Developer Command Prompt for VS 2022` command prompt and change to the NppModernShell folder.
+The following command expects the following:
+* The pfx certificate is called MyCert.pfx
+* The password for the pfx certificate is: `Test1234`
+
+Make the needed changes to match the real certificate.
+```
+SignTool.exe sign /fd SHA256 /tr http://timestamp.digicert.com /td sha256 /a /f .\MyCert.pfx /p Test1234 /d "Notepad++" /du https://notepad-plus-plus.org/ Packaging\NppModernShell.msix
```

Comment:
Rather:
`SignTool.exe sign /fd SHA256 /tr http://timestamp.digicert.com /td sha256 /a /f .\MyCert.pfx /p Test1234 /d "Notepad++" /du https://notepad-plus-plus.org/ NppModernShell.msix`
(without *Packaging*)

Reply:
Indeed, I have corrected and will push shortly